File: DSA/Sorting/BubbleSort.py
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        swapped = False
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True
        if not swapped:
            break


#example array
arr = [4, 5, 2, 3, 1]
bubble_sort(arr)

for num in arr:
    print(num, end=" ")

print()

File: DSA/C#/BubbleSort.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BubbleSort
{
    class Program
    {
        static void sort(int[] arr)
        {
            int n = arr.Length;
            bool swapped;

            do
            {
                swapped = false;

                for (int i = 1; i < n; i++)
                {
                    if (arr[i - 1] > arr[i])
                    {
                        // Swap the elements
                        int temp = arr[i - 1];
                        arr[i - 1] = arr[i];
                        arr[i] = temp;
                        swapped = true; 
                    }
                }
                n--;
            } while (swapped);
        }
        static void printArray(int[] array) { 
        foreach (int i in array){
        Console.WriteLine(i);
        }
        }
        static void Main(string[] args)
            {
                int[] arr = { 55, 44, 33, 22, 11 };
                sort(arr);
                Console.WriteLine("Sorted array:");
                printArray(arr);
                Console.ReadKey();
            }
    }
}
